warehouse/ingest-core/src/main/java/datawave/ingest/data/config/ingest/ErrorShardedIngestHelper.java

```diff
@@ -1,39 +1,284 @@
 
 import datawave.ingest.data.RawRecordContainer;
+import datawave.ingest.data.Type;
+import datawave.ingest.data.TypeRegistry;
 import datawave.ingest.data.config.NormalizedContentInterface;
 
 /**
  *
  */
 public class ErrorShardedIngestHelper extends BaseIngestHelper {
 
+    private static final Logger log = LoggerFactory.getLogger(ErrorShardedIngestHelper.class);
+
+    /**
+     * <p>
+     * Configuration inserted between the <datatype> and the <field-constant> for explicit error parsing.
+     * </p>
+     * <p>
+     * For example:
+     * </p>
+     * <p>
+     * {@code String target = <some-datatype> + DATATYPE_ERROR + INDEX_FIELDS;}
+     * </p>
+     */
+    private static final String ERROR = "error";
+    private IngestHelperInterface delegate = null;
+
+    private Map<Type,IndexedFields> errorIndexedFields = new HashMap<>();
+    private Map<Type,IndexedFields> errorReverseIndexedFields = new HashMap<>();
+
+    protected boolean hasErrorIndexDisallowlist = false;
+    protected boolean hasErrorReverseIndexDisallowlist = false;
+
+    private Type activeDataType;
+
+    private static class IndexedFields {
+        private Set<String> indexedFields = new HashSet<>();
+        private Map<String,Pattern> patterns = new HashMap<>();
+        private Set<String> unindexedFields = new HashSet<>();
+    }
+
+    /*
+     * SETH NOTE
+     *
+     * tests for the setup method
+     * https://github.com/NationalSecurityAgency/datawave/pull/2864/files#diff-86d9d0c6cfcff5b686be27e01ab927c38f6c347f59faeebdedd2ccbf96834d70 1. Verify if no
+     * global or datatype specific i/ri configs given, setup does not throw exception. 2. Verify if global i/ri given, setup does not throw exception. Verify
+     * datatype specific is still parsed. 3. Verify if global i/ri given, but not datatype specific, setup does not throw exception. 4. Verify that if both
+     * allow list and disallow list given for datatype specific, error is thrown.
+     */
+
     @Override
     public void setup(Configuration config) {
+
+        super.setup(config);
+
         // we are error
         config.set(Properties.DATA_NAME, "error");
-        super.setup(config);
+
+        // get all config properties that start with "error."
+        Map<String, String> errorProps = config.getPropsWithPrefix(ERROR);
+
+        // handle the index methods specified in the configuration properties for each datatype
+        for (var entry : errorProps.entrySet()){
+
+            String propertyKey = entry.getKey();
+
+            // get the property's datatype
+            // "error.<datatype>.<etc>..."
+            List<String> keyParts = List.of(propertyKey.split("\\."));
+            int datatypeIndex = keyParts.indexOf(ERROR) + 1;
+
+            String dataTypeString = keyParts.get(datatypeIndex);
+            String errorDataTypeString = ERROR + "." + dataTypeString;
+
+            activeDataType = TypeRegistry.getType(dataTypeString);
```

Comment:
We do not want to ever set `activeDataType` in the setup method, or use it when parsing out the property values. We want to avoid implying that the `activeDataType` should ever be set outside of the setter method for it. You should use a local variable instead to contain the parsed datatype.